crates/shared/src/sources/swapr.rs

```diff
@@ -22,13 +24,15 @@ const FEE_BASE: u32 = 10_000;
 impl PoolReading for SwaprPoolReader {
     fn read_state(&self, pair: TokenPair, block: BlockId) -> BoxFuture<'_, Result<Option<Pool>>> {
         let pair_address = self.0.pair_provider.pair_address(&pair);
-        let pair_contract = ISwaprPair::at(&self.0.web3, pair_address);
-
         let fetch_pool = self.0.read_state(pair, block);
-        let fetch_fee = pair_contract.swap_fee().block(block).call();
 
         async move {
-            let (pool, fee) = futures::join!(fetch_pool, fetch_fee);
+            // Inside the future because `.call` keeps a reference to the instance
```

Comment:
Probably not necessary to have this as a code comment. When somebody moves this line around the compiler will yell at them anyway, no?